tests/models/bridgetower/test_modeling_bridgetower.py

```diff
@@ -0,0 +1,414 @@
+        self.vocab_size = 50265
+        self.num_channels = 3
+        self.seq_length = 4
+        self.num_image_features = 325
+        self.batch_size = 1
+        self.is_training = False
+        self.image_size = image_size
+        self.expected_num_hidden_layers = 32
+        self.output_hidden_states = output_hidden_states
+
+    def prepare_config_and_inputs(self):
+        input_ids = ids_tensor([self.batch_size, self.seq_length], self.vocab_size)
+        attention_mask = random_attention_mask([self.batch_size, self.seq_length])
+
+        pixel_values = floats_tensor([self.batch_size, self.num_channels, self.image_size, self.image_size])
+        pixel_mask = random_attention_mask([self.batch_size, self.image_size, self.image_size])
+
+        config = self.get_config()
+
+        return (config, input_ids, attention_mask, pixel_values, pixel_mask)
+
+    def get_config(self):
+        return BridgeTowerConfig(
+            share_cross_modal_transformer_layers=self.share_cross_modal_transformer_layers,
+            drop_rate=self.drop_rate,
+            head_hidden_scale=self.head_hidden_scale,
+            hidden_act=self.hidden_act,
+            hidden_size=self.hidden_size,
+            initializer_factor=self.initializer_factor,
+            image_size=self.image_size,
+            is_encoder_decoder=self.is_encoder_decoder,
+            layer_norm_eps=self.layer_norm_eps,
+            share_link_tower_layers=self.share_link_tower_layers,
+            link_tower_type=self.link_tower_type,
+            num_attention_heads=self.num_attention_heads,
+            num_hidden_layers=self.num_hidden_layers,
+            tie_word_embeddings=self.tie_word_embeddings,
+            init_layernorm_from_vision_encoder=self.init_layernorm_from_vision_encoder,
+            output_hidden_states=self.output_hidden_states,
+        )
+
+    def create_and_check_model(
+        self,
+        config,
+        input_ids,
+        attention_mask,
+        pixel_values,
+        pixel_mask,
+    ):
+        model = BridgeTowerModel(config=config)
+        model.to(torch_device)
+        model.eval()
+
+        result = model(input_ids, attention_mask=attention_mask, pixel_values=pixel_values, pixel_mask=pixel_mask)
+        result = model(input_ids, attention_mask=attention_mask, pixel_values=pixel_values)
+        self.parent.assertEqual(result["text_features"].shape, (self.batch_size, self.seq_length, self.hidden_size))
+        self.parent.assertEqual(
+            result["image_features"].shape, (self.batch_size, self.num_image_features, self.hidden_size)
+        )
+        self.parent.assertEqual(result["pooler_output"].shape, (self.batch_size, 2 * self.hidden_size))
+
+    def create_and_check_for_image_and_text_retrieval(
+        self,
+        config,
+        input_ids,
+        attention_mask,
+        pixel_values,
+        pixel_mask,
+    ):
+        bridgeTowerITMHead_output_last_dimension = 2
```

Comment:
Please use a lowercased variable name, with words separated by underscores.

Reply:
We have fixed this in our latest commit. Thanks for the feedback!